File: TopInterview150/TwoPointers/isSubsequence/solution.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) == 0:
            return True
        if len(t) == 0:
            return False
        p1 = 0
        p2 = 0
        while p2 < len(t):
            if (s[p1] == t[p2]):
                p1+=1
                if p1 == len(s):
                    return True
            p2+=1
        return False

File: TopInterview150/TwoPointers/isSubsequence/solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.length() == 0) return true;
        if (t.length() == 0) return false;
        int p1 = 0; int p2 = 0;
        while (p2 < t.length())
        {
            if (s[p1] == t[p2])
            { 
                p1++;
                if (p1 == s.length())
                {
                    return true;
                }
            }
            p2++;
        }
        return false;
    }
};